fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1331,10 +1431,12 @@ ACTOR Future<Void> reply(CommitBatchContext* self) {
 	// self->committedVersion by reporting commit version first before updating self->committedVersion. Otherwise, a
 	// client may get a commit version that the master is not aware of, and next GRV request may get a version less than
 	// self->committedVersion.
+
 	TEST(pProxyCommitData->committedVersion.get() > self->commitVersion); // later version was reported committed first
+
 	if (self->commitVersion >= pProxyCommitData->committedVersion.get()) {
 		state Optional<std::set<Tag>> writtenTags;
-		if (SERVER_KNOBS->ENABLE_VERSION_VECTOR) {
+		if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
 			writtenTags = self->writtenTags;
```

Comment:
Can we add an assert here that "self->writtenTags == self->writtenTagsPreResolution"?

Reply:
Ok

Reply:
Thinking more about it, I think this assert will be valid only when "self->hasMetadataMutation == false"?

Reply:
the way the assert is currently written, I verify every member of the post resolution set of tLogs is a member of the pre-resolution set of tLogs. The illegal case I want to check is if we try to use a tLog for which we did not get a PCV. This passes when self->hasMetadataMutation == true.

Reply:
Ok.